src/domain/agent/agent/agent.service.ts

```diff
@@ -136,18 +136,39 @@ export class AgentService {
     return agents;
   }
 
+  private getAgentCacheKey(agentId: string): string {
+    return `@agent:id:${agentId}`;
+  }
+
+  private async getAgentFromCache(id: string): Promise<IAgent | undefined> {
+    return await this.cacheManager.get<IAgent>(this.getAgentCacheKey(id));
+  }
+
+  private async setAgentCache(agent: IAgent): Promise<IAgent> {
+    const cacheKey = this.getAgentCacheKey(agent.id);
+    return await this.cacheManager.set<IAgent>(cacheKey, agent, {
+      ttl: 60,
```

Comment:
This seems a bit short, make longer?

Reply:
Increased it to 15 minutes and is configurable now.